go/libraries/doltcore/doltdb/doltdb.go

```diff
@@ -235,6 +235,17 @@ func (ddb *DoltDB) Close() error {
 	return ddb.db.Close()
 }
 
+// ParseHashString converts a string representing a hash into a hash.Hash.
+func ParseHashString(hashStr string) (hash.Hash, error) {
```

Comment:
This function is also called by DoltDB.Resolve. It's not wholly new, it's a refactor of getCommitValForHash, which this PR removes.

Reply:
I missed that context, thanks.

I think (I'm reasonably certain) that there's no valid use case for passing this function a "#" at this level. That's strictly a command-line parsing issue (and I don't think there's any command line that accepts a hash beginning with "#"). It's very old code and lot has changed since it was written.

Since it's called in only one place, just inline it and delete the method. Then move the "#" stripping functionality to your command if you want to keep it, up to you.

Reply:
Done.